vars/buildChain.groovy

```diff
@@ -0,0 +1,25 @@
+import groovy.json.JsonSlurper
+import org.yaml.snakeyaml.Yaml
+
+/*
+* It gets buildChain verion from composite action.yml file
+*/
+def getBuildChainVersionFromCompositeActionFile(String actionFilePath = '.ci/actions/build-chain/action.yml', String usesContainingString = 'github-action-build-chain@') {
+    def actionFileContent = readFile actionFilePath
+    def actionObject = new Yaml().load(actionFileContent)
+
+    def uses = actionObject.runs.steps.uses
+    def action = uses != null ? uses.find({ it.contains(usesContainingString) }) : null
+    if (action == null) {
+        throw new RuntimeException("There's not steps with 'uses' for build-chain ${usesContainingString}")
+    }
+
+    def buildChainScmRevision = action.substring(action.indexOf('@') + 1)
+    try {
+        def url = new URL("https://raw.githubusercontent.com/kiegroup/github-action-build-chain/${buildChainScmRevision}/package.json")
+        def packageJson = url.getText()
+        return new JsonSlurper().parseText(packageJson).version
```

Comment:
Can you explain this last part of the method?
Isn't the version already the `buildChainScmRevision` value? Don't the tag/branch corresponds with the version of action?
I was expecting that parsing the URL was enough without the need to download and parse  `package.json` 🤔

Reply:
this is a very good question @danielezonca 
well.... the thing is `uses` from github actions jobs/composite actions files point to scmRevision from repository (at the end they checkout the project and runs whatever it is define in `run/main` field from action.yml https://github.com/kiegroup/github-action-build-chain/blob/baad995df06d755949ed6c80f9e9b54e346b3e01/action.yml#L25
We are using npm library from Jenkins and it could be the case the scmRevision from composite action and npm library version are not aligned (like it is the case). We always increase npm version whenever anything changes in build-chain library but we push a release (new tag) whenever we consider it is something important for GHA.
Imagine we change CLI arguments for the tool, it does not make sense to release anything for GHA.

is it clear enough?

Reply:
Thanks for the explanation 👍 
Given that do a release is a very cheap action I would prefer to do an "official" release (even minor) every time we expect to use it for all our pipelines, otherwise it is very hard in GHA file to understand what is happening because the URL is always `...@v2.3.4` while maybe in reality it is using a newer (unreleased) version.
Wdyt?
It is quite unrelated to this PR, it is more a general discussion.

> Imagine we change CLI arguments for the tool, it does not make sense to release anything for GHA.

This is true but how often do we expect this to happen? If CLI will change way more times compared to the actual action we might consider to version/release it differently.

Reply:
@danielezonca ok, we can agree or disagree on how to deal with BC releases/versioning (and I'm more about aligning versions as you suggest than to keep the current strategy) but... don't you think I'm complicating things here a bit? :thinking: what if we keep two inputs in the action like
```
  scm-revision:
    description: "The scm revision to get the repository from"
    default: "@v2.3.4"
    required: false
  npm-version:
    description: "The npm library version to get it from Jenkins"
    default: ''2.3.4""
    required: false
```

this way we won't need to get version from package.json and so on... it's more clear, the `getBuildChainVersionFromCompositeActionFile` will be less prone to fail for such simple thing, we can use the release/version strategy we want (less coupled) and we can probably want to use different versions for GHA/Jenkins for any reason...
wdyt?

Reply:
I have no strong opinion about the possibility or not to override the version. I think we should try to have GHA and Jenkins always aligned and updated together so that we avoid situation like errors because Jenkins executor are still updating the version.
In a nutshell I think:
- URL should be enough to identify a version (so no `package.json` parsing)
- it should be nice to have version in a property but it is more important that both Jenkins and GHA reads from the same source so if it is necessary to split the URL fine for me
- Having the possibility to have override the version could be a nice to have especially for tests but it should be ~never used in the "stable"/"main" setup